### ISSUE TITLE
[ISSUE] Read-only root filesystem issues on operate/tasklist when multitenancy is enabled

### ISSUE THREAD

**Issue by jessesimpson36** [camunda/camunda-platform-helm#954]
**Describe the issue:**

When enabling multitenancy with the following parameters set:
```
global:
  multitenancy:
    enabled: true

identity:
  postgresql:
    enabled: true
```
Operate and tasklist  fail to come up healthily due to a failure to write to `/usr/local/operate/.optimize`

```
java.nio.file.FileSystemException: /usr/local/operate/.camunda: Read-only file system                                                                           
    at sun.nio.fs.UnixException.translateToIOException(Unknown Source) ~[?:?]                                                                                   
    at sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source) ~[?:?]                                                                                     
    at sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source) ~[?:?]                                                                                     
    at sun.nio.fs.UnixFileSystemProvider.createDirectory(Unknown Source) ~[?:?]                                                                                 
    at java.nio.file.Files.createDirectory(Unknown Source) ~[?:?]                                                                                               
    at java.nio.file.Files.createAndCheckIsDirectory(Unknown Source) ~[?:?]                                                                                     
    at java.nio.file.Files.createDirectories(Unknown Source) ~[?:?]                                                                                             
    at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsCache.ensureCacheFileExists(OAuthCredentialsCache.java:147) ~[zeebe-client-java-8.3.0.jar!/:8.3.0]    
    at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsCache.writeCache(OAuthCredentialsCache.java:74) ~[zeebe-client-java-8.3.0.jar!/:8.3.0]                
    at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsProvider.lambda$shouldRetryRequest$0(OAuthCredentialsProvider.java:110) ~[zeebe-client-java-8.3.0.jar!
    at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsCache.withCache(OAuthCredentialsCache.java:111) ~[zeebe-client-java-8.3.0.jar!/:8.3.0]                
    at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsProvider.shouldRetryRequest(OAuthCredentialsProvider.java:106) ~[zeebe-client-java-8.3.0.jar!/:8.3.0] 
    at io.camunda.zeebe.client.impl.RetriableClientFutureImpl.onError(RetriableClientFutureImpl.java:48) [zeebe-client-java-8.3.0.jar!/:8.3.0]                  
    at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:481) [grpc-stub-1.58.0.jar!/:1.58.0]                               
    at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:567) [grpc-core-1.58.0.jar!/:1.58.0]                                                   
    at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:71) [grpc-core-1.58.0.jar!/:1.58.0]                                                       
    at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:735) [grpc-core-1.58.0.jar!/:1.58.0]              
    at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:716) [grpc-core-1.58.0.jar!/:1.58.0]             
    at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) [grpc-core-1.58.0.jar!/:1.58.0]                                                            
    at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) [grpc-core-1.58.0.jar!/:1.58.0]                                                   
    at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]                                                                                  
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]                                                                                 
    at java.lang.Thread.run(Unknown Source) [?:?]                

```

**Actual behavior:**

**Expected behavior:**

Applications start up healthily.

**How to reproduce:**

**Logs:**

**Environment:**

**Please note: Without the following info, it's hard to resolve the issue and probably it will be closed.**

- Platform: KIND
- Helm CLI version: 3.11
- Chart version: 8.3.0
- Values file: [total.yaml.txt](https://github.com/camunda/camunda-platform-helm/files/12884127/total.yaml.txt)

Support: [SUPPORT-18783](https://jira.camunda.com/browse/SUPPORT-18783)

**Comment by lukas-beumer** [comment]
The problem does not only exist with multitenancy. We have had the problem as well, but then solved it with an emptyDir.

```
- name: operate-dir
  mountPath: /usr/local/operate/.optimize
```

**Comment by jessesimpson36** [comment]
During the support ticket investigation we learned of 2 issues. One which was the stacktrace mentioned above. we believe in the context of the support ticket, that was merely noise, and when multitenancy is disabled, the application can recover successfully despite this error. 

When multi-tenancy is enabled, we get a warning
```
i.c.z.c.i.ZeebeCallCredentials           : The request's security level does not guarantee that the credentials will be confidential.
```
which repeats itself at a fast rate, repeatedly. and neither Operate nor tasklist can come up successfully once it starts looping on that warning.

Application teams will be following up on the fatal repeating warning, and our helm chart primarily needs to 
 either set  ZEEBE_CLIENT_CONFIG_PATH  to /tmp/anyfilename  or mount the $home/.camunda folder as a volume as @lukas-beumer mentioned.   

I did confirm that in either scenario, whether multitenancy is enabled or disabled, that there are no contents in `.camunda` until authentication, which means we do not need an initContainer to copy existing files into the volume.

**Comment by lukas-beumer** [comment]
In our environment, however, we do not use the multitenancy function at all. The error floods our logs anyway and Tasklist and Operate won't start. We use an external keycloak, if that is important.

If I set the authentication mode in the Zeebe gateway to none, everything starts up cleanly.

I'll try that with the ZEEBE_CLIENT_CONFIG_PATH, even the emptyDir didn't help. The error no longer occurred, but the container was still not ready.

So it is definitely due to the authentication at the Zeebe gateway. It's not really round yet.

**Comment by lukas-beumer** [comment]
@jessesimpson36 

Unfortunately it didn't work, in the DEBUG log I still get an error in both containers despite environment variable. Here are the errors from Kubernetes:

> 2023-10-16 05:35:59.344  WARN 7 --- [ault-executor-3] i.c.z.c.i.ZeebeCallCredentials           : The request's security level does not guarantee that the credentials will be confidential.
2023-10-16T05:35:59.345651204Z 2023-10-16 05:35:59.345 DEBUG 7 --- [-worker-ELG-1-1] i.g.n.NettyClientHandler                 : [id: 0x7d89e1e8, L:/100.65.7.199:57278 - R:camunda-platform-zeebe-gateway/100.65.125.193:26500] OUTBOUND HEADERS: streamId=13101 headers=GrpcHttp2OutboundHeaders[:authority: camunda-platform-zeebe-gateway:26500, :path: /gateway_protocol.Gateway/Topology, :method: POST, :scheme: http, content-type: application/grpc, te: trailers, user-agent: zeebe-client-java/8.3.0 grpc-java-netty/1.58.0, grpc-accept-encoding: gzip, authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJ0WVJkXzBZTlVQRC00RG5jS1pSREJqU00tMFV0RTdwLVRETnZWVWVtNXA4In0.eyJleHAiOjE2OTc0MzU3NTksImlhdCI6MTY5NzQzNDU1OSwianRpIjoiNTU4ZDcwYTMtM2UzOS00NThjLTg4NTYtMjQ1ZDFmMzBmMmU5IiwiaXNzIjoiaHR0cHM6Ly9pYW0tc3RhZ2UuaWFtLnBhYXMucmV3ZS5sb2NhbC9hdXRoL3JlYWxtcy9wcm9kMTkyLWNhbXVuZGEiLCJhdWQiOlsiemVlYmUiLCJ6ZWViZS1hcGkiLCJhY2NvdW50Il0sInN1YiI6IjQ5OTIwYTZkLWVlYzItNDVhOS1hYzhiLTU0NTE4YWNkMjk3MSIsInR5cCI6IkJlYXJlciIsImF6cCI6InplZWJlIiwiYWNyIjoiMSIsInJlYWxtX2FjY2VzcyI6eyJyb2xlcyI6WyJvZmZsaW5lX2FjY2VzcyIsImRlZmF1bHQtcm9sZXMtcHJvZDE5Mi1jYW11bmRhIiwidW1hX2F1dGhvcml6YXRpb24iXX0sInJlc291cmNlX2FjY2VzcyI6eyJ6ZWViZS1hcGkiOnsicm9sZXMiOlsid3JpdGU6KiJdfSwiYWNjb3VudCI6eyJyb2xlcyI6WyJtYW5hZ2UtYWNjb3VudCIsIm1hbmFnZS1hY2NvdW50LWxpbmtzIiwidmlldy1wcm9maWxlIl19fSwic2NvcGUiOiJwcm9maWxlIGVtYWlsIiwiY2xpZW50SG9zdCI6IjEwLjE0OS40MC4yMjMiLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsInBlcm1pc3Npb25zIjp7InplZWJlLWFwaSI6WyJ3cml0ZToqIl0sImFjY291bnQiOlsibWFuYWdlLWFjY291bnQiLCJtYW5hZ2UtYWNjb3VudC1saW5rcyIsInZpZXctcHJvZmlsZSJdfSwicHJlZmVycmVkX3VzZXJuYW1lIjoic2VydmljZS1hY2NvdW50LXplZWJlIiwiY2xpZW50QWRkcmVzcyI6IjEwLjE0OS40MC4yMjMiLCJjbGllbnRfaWQiOiJ6ZWViZSJ9.EA9vqBLNyFzWZSmX2Qyxh-iCsqUSseW68W6ESCgTYrupeeS43lY0nMK_zfmBxSBAD_T8GXw9uCCcWmRhvvT9_GcJJ6OUsWPD9nq_2Ar
ngmiy_LBunYCIxrKd_nZnBe-PLAyzMdLcQhcUHrpxMia6JaRGnMVgptUnmk4J-k4qcziJNNJ3DGvJvkGMe0pcMjuL9Jxz7oDrSmerKIdjurWeHDy00zFIS3InqwlhFc0m8JBsZN4LOA9bNjjjmQwsYWT0MOA3PxIRb94h5WRH3GSbN_oz8ZJphCdbkZTd53McbM2nt57M05PQzLaG5prY5Xq5DQ8Rd7PL-MoFrrun_JeWzw, grpc-timeout: 9999632u] streamDependency=0 weight=16 exclusive=false padding=0 endStream=false
2023-10-16 05:35:59.345 DEBUG 7 --- [-worker-ELG-1-1] i.g.n.NettyClientHandler                 : [id: 0x7d89e1e8, L:/100.65.7.199:57278 - R:camunda-platform-zeebe-gateway/100.65.125.193:26500] OUTBOUND DATA: streamId=13101 padding=0 endStream=true length=5 bytes=0000000000
2023-10-16 05:35:59.345 DEBUG 7 --- [-worker-ELG-1-1] i.g.n.NettyClientHandler                 : [id: 0x7d89e1e8, L:/100.65.7.199:57278 - R:camunda-platform-zeebe-gateway/100.65.125.193:26500] INBOUND HEADERS: streamId=13101 headers=GrpcHttp2ResponseHeaders[:status: 200, content-type: application/grpc, grpc-status: 16, grpc-message: Failed to parse bearer token, see cause for details] padding=0 endStream=true
2023-10-16T05:35:59.346791624Z 2023-10-16 05:35:59.346 DEBUG 7 --- [ault-executor-3] s.n.w.p.h.HttpURLConnection              : sun.net.www.MessageHeader@676e38907 pairs: {POST /auth/realms/camunda/protocol/openid-connect/token HTTP/1.1: null}{Content-Type: application/x-www-form-urlencoded}{Accept: application/json}{User-Agent: zeebe-client-java/8.3.0}{Host: [replaced]}{Connection: keep-alive}{Content-Length: 89}
2023-10-16 05:35:59.365 DEBUG 7 --- [ault-executor-3] s.n.w.p.h.HttpURLConnection              : sun.net.www.MessageHeader@3148e34414 pairs: {null: HTTP/1.1 200 OK}{Server: nginx}{Date: Mon, 16 Oct 2023 05:35:59 GMT}{Content-Type: application/json}{Content-Length: 1627}{Connection: keep-alive}{Referrer-Policy: no-referrer}{X-Frame-Options: SAMEORIGIN}{Strict-Transport-Security: max-age=31536000; includeSubDomains}{Cache-Control: no-store}{X-Content-Type-Options: nosniff}{Set-Cookie: KC_RESTART=; Version=1; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/camunda/; Secure; HttpOnly}{Pragma: no-cache}{X-XSS-Protection: 1; mode=block}

**Comment by jessesimpson36** [comment]
@lukas-beumer ,  There was a few different issues with 8.3.0 and I think you encountered some of the other issues that we ran into and fixed for 8.3.1.  

Please try upgrading to 8.3.1 and if there's any issues you're having, you can open a github issue. For this ticket though, I've already tested and I do not get the stacktrace I got at the beginning of this ticket, so I will mark it as completed.

Somewhere in my debugging, I did run into " i.c.z.c.i.ZeebeCallCredentials : The request's security level does not guarantee that the credentials will be confidential." being listed repeatedly dozens of times per second, but we fixed that in 8.3.1. I'm not sure if that's related to what you encountered.
